src/io/perun.clj

```diff
@@ -481,28 +563,22 @@
    If permalink ends in slash, index.html is used as filename.
    If permalink is not set, the original filename is used with file extension set to html."
   [o out-dir  OUTDIR   str  "the output directory (default: \"public\")"
-   _ filterer FILTER   code "predicate to use for selecting entries (default: `:content`)"
+   _ filterer FILTER   code "predicate to use for selecting entries (default: `:has-content`)"
    r renderer RENDERER sym  "page renderer (fully qualified symbol which resolves to a function)"
    m meta     META     edn  "metadata to set on each entry"]
   (let [options (merge +render-defaults+ *opts*)]
     (letfn [(render-paths [fileset options]
               (let [entries (filter (:filterer options) (pm/get-meta fileset))
                     paths (reduce
-                           (fn [result {:keys [path] :as entry*}]
-                             (let [entry (merge meta entry*)
-                                   render-data   {:meta    (pm/get-global-meta fileset)
-                                                  :entries (vec entries)
-                                                  :entry   entry}
-                                   page-filepath (perun/create-filepath
-                                                  (:out-dir options)
-                                                  ; If permalink ends in slash, append index.html as filename
-                                                  (or (some-> (:permalink entry)
-                                                              (string/replace #"/$" "/index.html")
-                                                              perun/url-to-path)
-                                                      (string/replace path #"(?i).[a-z]+$" ".html")))]
+                           (fn [result {:keys [path permalink] :as entry}]
+                             (let [content (slurp (boot/tmp-file (boot/tmp-get fileset path)))
+                                   new-path (make-path (:out-dir options) permalink path)
+                                   meta-entry (merge meta entry)
+                                   content-entry (assoc meta-entry :content content)]
                                (perun/report-debug "render" "rendered page for path" path)
```

Comment:
This logging message is potentially confusing, as the rendering isn't actually occurring here.  When someone's render task is crashing they might think that it had worked given that this printed out for all files.  Perhaps it should be moved to render-to-paths.